docs/adr/ADR-001-Theme_and_tokens.md

```diff
@@ -0,0 +1,87 @@
+# ADR-001: Theme and tokens
+
+## Date
+
+2023-10-04
+
+## Status
+
+PROPOSED
+
+## Context
+
+At the moment, when providing custom themes to the Design System, the user can only override the palette of colors.
+In some cases, this can be limiting, as the user may need to override specific parts of components to respect A11Y
+guidelines or to have more coherent colors.
+
+## Decision
+
+We will change the structure of the Theme object provided to the Design System to allow providing overrides for either
+the palette or component tokens.
+Any change in the palette will be reflected in the component tokens.
+Component tokens will accept only colors from the palette.
+At the moment, only colors can be overridden, but this could be extended to other tokens in the future.
+
+The input structure accepted by the `DesignSystem` provider would be as follows:
+
+```typescript
+interface ThemeCustomization {
+    palette: {
+        brand-<tint>: string; // e.g: brand-05: #E0F0F9
+        accent-<tint>: string;
+        neutral-<tint>: string;
+        alert-<tint>: string;
+        informative-<tint>: string;
+        success-<tint>: string;
+        warning-<tint>: string;
+    };
+    components: {
+        <component>-<flavor?>-<property>: <palette-token>;
```

Comment:
```suggestion
        <component>-<flavor?>-<property>: <ref-token>;
```
In order to allow values from other ref tokens (in the future?), we might not want to restrict this definition to only palette colors.

If we go that route, we could flatten the ref object to prevent having to search multiple keys for overrides. That would give us a `ThemeCustomization` interface that would be a strict subset of the `Theme` interface.

```typescript
interface ThemeCustomization {
    ref: {
        'color-brand-50': string;
        'font-weight-bold': string;
        // ...
    };
    component: {
        'button-primary-background-color': 'color-brand-50';
        'button-primary-text-hover': 'font-weight-bold';
        // ...
    };
}
```

It would remain possible to restrict the type of a component token to a specific list of ref tokens if this is something we want to enforce:
```typescript
interface Palette {
    'color-brand-50': string;
}

interface TextAttributes {
    'font-weight-bold': string;
}

type RefTokens = Palette & TextAttributes;

interface ComponentTokens {
    'button-primary-background-color': keyof Palette;
    'button-primary-text-hover': keyof TextAttributes;
}

interface Theme {
    ref: RefTokens;
    component: ComponentTokens;
}
```